backend/app/security.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any
+
+import fastapi
+import requests_cache
+from fastapi import security
+from jose import exceptions, jwt
+
+from app import config
+
+settings = config.get_settings()
+
+_oauth_scheme = security.OAuth2AuthorizationCodeBearer(
+    authorizationUrl=f"{settings.auth_server[0]}/protocol/openid-connect/auth",
+    tokenUrl=f"{settings.auth_server[0]}/protocol/openid-connect/token",
+)
+
+
+_auth_server_exception = fastapi.HTTPException(
+    status_code=500, detail="Could not get JWK set from auth server"
+)
+_invalid_token_exception = fastapi.HTTPException(
+    status_code=401,
+    detail="Invalid token",
+    headers={"WWW-Authenticate": "Bearer"},
+)
+
+
+def _get_jwks(url: str, clear_cache: bool = False) -> Any:
+    try:
```

Comment:
should this all be in the try block? If so, it might be helpful to log the exception

Reply:
What would you take out from the try block?
Logging is a good point whether we keep it like this or change it. I haven't configured logging at all in general. I plan to do that in an upcoming PR.

Reply:
I dunno actually. If you want to catch all the potential errors from all the lines (which I think it's the case) then we should keep it like this. Otherwise, if one single line, then an else block and a finally where we close the context would be safer. But I think we want to catch all the lines, right?

Reply:
The 'if clear_cache` block could be outside the try block, but the rest we want to cover.

Reply:
ok, what about: 
```python
session = requests_cache.CachedSession(
            backend="memory", expire_after=3600, cache_control=True
        )
if clear_cache:
  session.cache.clear()
try: 
  response = session.get(url=url)
  payload = response.json()
  response = session.get(url=payload["jwks_uri"])
  jwks = response.json()
except Exception:
  raise _auth_server_exception
finally: 
  session.close()
```